extensions/dataref.md

```diff
@@ -0,0 +1,73 @@
+# Dataref (Claim Check Pattern)
+
+As defined by the term [Data](spec.md#data), CloudEvents MAY include domain-specific
```

Comment:
CI failure noticed that this needs to be `../spec` since we're in the `extensions` dir now.  I think there are a few more below too.